lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1551,8 +1551,13 @@ class BuildControl : public calyx::FuncOpPartialLoweringPattern {
             rewriter.create<calyx::SeqOp>(ifOp.getThenRegion().getLoc());
         auto *thenSeqOpBlock = thenSeqOp.getBodyBlock();
 
-        rewriter.setInsertionPointToEnd(thenSeqOpBlock);
+        auto *thenBlock = &ifOp.getThenRegion().front();
+        LogicalResult res = buildCFGControl(path, rewriter, thenSeqOpBlock,
```

Comment:
Just to make sure I'm understanding correctly: your code is only supporting one level of nesting of `if` statements?

Reply:
Arbitrary level because `buildCFGControl` is a recursive function

Reply:
Ah got it, so initially it was just ignoring the `then` and `else` blocks of the `if` control.